action.yaml

```diff
@@ -155,51 +155,17 @@ runs:
       shell: bash
       run: ${GITHUB_ACTION_PATH}/locate_trunk.sh
 
-    - name: Init on-demand
-      shell: bash
-      run: |
-        if [ ! -e .trunk/trunk.yaml ]; then
-          ${TRUNK_PATH} init
-        fi
-
-    - name: Detect npm/yarn/pnpm
+    - name: Detect
       id: detect
       shell: bash
       run: |
-        if [ -e package-lock.json ]; then
-          echo "package_manager=npm" >> $GITHUB_OUTPUT
-          echo "install_cmd=npm ci" >> $GITHUB_OUTPUT
-          echo "hash_glob=**/package-lock.json" >> $GITHUB_OUTPUT
-        elif [ -e yarn.lock ]; then
-          echo "package_manager=yarn" >> $GITHUB_OUTPUT
-          echo "install_cmd=yarn install --immutable" >> $GITHUB_OUTPUT
-          echo "hash_glob=**/yarn.lock" >> $GITHUB_OUTPUT
-        elif [ -e pnpm-lock.yaml ]; then
-          echo "package_manager=pnpm" >> $GITHUB_OUTPUT
-          echo "install_cmd=pnpm install --frozen-lockfile" >> $GITHUB_OUTPUT
-          echo "hash_glob=**/pnpm-lock.yaml" >> $GITHUB_OUTPUT
+        if [ ! -e .trunk/setup-ci ]; then
+          mkdir -p .trunk
+          ln -s ${{ github.action_path }}/setup-env .trunk/setup-ci
```

Comment:
2 quick points here:

1. Is the expectation that the user's `.trunk/setup-ci` will include _all_ setup (including init'ing trunk if necessary). Do we want to document that somewhere, or
2. Is this part of a user request or to support future SAAS work?
3. Perhaps most importantly with regard to functionality, do we need to git ignore/trunk ignore the `.trunk/setup-ci` folder? Won't that get triggered as a change during the `trunk check` run?

Tangentially, after this goes in I should probably update the upgrade action to include this flow.

Reply:
1. Yes, and we'll need to document this
2. Support SaaS work - specifically we need it for feature parity with our current setup
3. That's a really good point - thanks for calling that out